authentication/src/main/java/io/camunda/authentication/config/WebSecurityConfig.java

```diff
@@ -26,32 +33,62 @@
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Primary;
 import org.springframework.context.annotation.Profile;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
 import org.springframework.security.access.expression.method.MethodSecurityExpressionHandler;
+import org.springframework.security.config.Customizer;
 import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.core.Authentication;
 import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
 import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+import org.springframework.security.web.csrf.CsrfFilter;
+import org.springframework.security.web.csrf.CsrfTokenRepository;
+import org.springframework.security.web.util.matcher.RequestMatcher;
 
 @Configuration
 @EnableWebSecurity
 @EnableMethodSecurity
-@Profile("auth-basic|auth-oidc")
+@Profile("auth-basic|auth-basic-with-unprotected-api|auth-oidc")
 public class WebSecurityConfig {
-  public static final String[] UNAUTHENTICATED_PATHS =
-      new String[] {
-        "/login",
-        "/logout",
-        // endpoint for failure forwarding
-        "/error",
-        // all actuator endpoints
-        "/actuator/**",
-        // endpoints defined in BrokerHealthRoutes
-        "/ready",
-        "/health",
-        "/startup"
-      };
+  public static final String SPRING_SECURITY_OAUTH_2_RESOURCESERVER_JWT_ISSUER_URI =
+      "spring.security.oauth2.resourceserver.jwt.issuer-uri";
+  public static final String SPRING_SECURITY_OAUTH_2_RESOURCESERVER_JWT_JWK_SET_URI =
+      "spring.security.oauth2.resourceserver.jwt.jwk-set-uri";
```

Comment:
Yes, I'll remove them. I had started moving more configuration over because I wanted to delete the old filter chains for tasklist and operate, but then decided to keep that for later.